MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/FingerDrawActivity.java

```diff
@@ -0,0 +1,376 @@
+  }
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_finger_drag_draw);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(@NonNull final MapboxMap mapboxMap) {
+        mapboxMap.setStyle(Style.LIGHT, new Style.OnStyleLoaded() {
+          @Override
+          public void onStyleLoaded(@NonNull Style style) {
+
+            FingerDrawActivity.this.mapboxMap = mapboxMap;
+
+            if (showSearchDataLocations) {
+              new LoadGeoJson(FingerDrawActivity.this).execute();
+            } else {
+              setUpExample(null);
+            }
+
+            findViewById(R.id.clear_map_for_new_draw_fab)
+              .setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+
+                  // Reset ArrayLists
+                  polygonList = new ArrayList<>();
+                  freehandDrawLineLayerPointList = new ArrayList<>();
+
+                  // Add empty Feature array to the sources
+                  drawLineSource = mapboxMap.getStyle().getSourceAs(FREEHAND_DRAW_LINE_LAYER_SOURCE_ID);
+                  if (drawLineSource != null) {
+                    drawLineSource.setGeoJson(FeatureCollection.fromFeatures(new Feature[]{}));
+                  }
+
+                  fillPolygonSource = mapboxMap.getStyle().getSourceAs(FREEHAND_DRAW_FILL_LAYER_SOURCE_ID);
+                  if (fillPolygonSource != null) {
+                    fillPolygonSource.setGeoJson(FeatureCollection.fromFeatures(new Feature[]{}));
+                  }
+
+                  // Reset camera position to default location
+                  mapboxMap.easeCamera(CameraUpdateFactory
+                    .newCameraPosition(new CameraPosition.Builder()
+                      .target(new LatLng(35.087497, -106.651261))
+                      .zoom(11.679132)
+                      .tilt(0)
+                      .bearing(0)
+                      .build()));
+
+                  enabledMapDrawing();
+                }
+              });
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Enable moving the map
+   */
+  private void enableMapMovement() {
+    mapView.setOnTouchListener(new View.OnTouchListener() {
+      @Override
+      public boolean onTouch(View view, MotionEvent motionEvent) {
+        return false;
+      }
+    });
+    mapboxMap.getUiSettings().setAllGesturesEnabled(true);
```

Comment:
I don't think that `setAllGesturesEnabled` invocations are needed in this example - as soon as you set a custom touch listener you're going to intercept all motion events anyway.

Reply:
yep, works fine with those two invocations removed.